### PR TITLE
Add an optional description field to cmake-kits.json file

### PR DESCRIPTION
**Pull request by TisziV**
### This changes visible behavior

The following changes are proposed:
- Add an optional description field to cmake-kits.json file.
- This description will appear in the kit selection menu instead of the default "Kit for toolcain file ...".

## The purpose of this change

In some projects there are many architectures to build for, and it is not always straightforward which kit should be used for which platform.
Hints helping with eg. product or platform names can be added to the description text to disambiguate.

### CODE REVIEW

**Comment by TisziV** [comment]
@microsoft-github-policy-service agree

File: src/kit.ts [hunk 1]
```diff
@@ -1093,6 +1098,9 @@ export async function scanForKitsIfNeeded(project: CMakeProject): Promise<boolea
  */
 export async function descriptionForKit(kit: Kit, shortVsName: boolean = false): Promise<string> {
     if (kit.toolchainFile) {
+        if (kit.description) {
```
Comment (bobbrow) [inline src/kit.ts:1101]: This should be moved outside (and above) the branch for `kit.toolchainFile`

**Review by bobbrow — CHANGES_REQUESTED** [review]

**Review by elahehrashedi — APPROVED** [review]

File: src/kit.ts (hunk 1 above)
Comment (TisziV) [reply src/kit.ts]: Thanks for the comment.
Added it to later commit.
Also adjusted kits.md accordingly.

**Comment by elahehrashedi** [comment]
@bobbrow adding description seems fine:

image

**Review by bobbrow — APPROVED** [review]
@elahehrashedi will fix the changelog and then this is good to go.

**Pull request merged**